### PR TITLE
(Cardigann v4) (Usenet)

### PR DESCRIPTION
**Pull request by bakerboy448**
Add Usenet Indexers

* Closes #116
* Closes #106

### CODE REVIEW

File: definitions/v4/nzbgeek.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+protocol: usenet
+encoding: UTF-8
+links:
+  - https://api.nzbgeek.info
+
+caps:
+  # These can typically be found at {site}/api?t=caps
+  categorymappings:
+    - { id: 1000, cat: Console, desc: Console }
+    - { id: 1010, cat: Console/NDS, desc: Console/NDS }
```
Comment (Qstick) [inline definitions/v4/nzbgeek.yml:17]: You can skip adding desc to all these I believe, it’ll just use standard desc for the mapped cat

Comment (bakerboy448) [reply definitions/v4/nzbgeek.yml:17]: Trivial to add (all +- semi-automated and parsed from the Caps Call). And the description is not consistent for all.  But I do believe geek is fine

**Comment by bakerboy448** [comment]
To Fix
```
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1035: Console/Switch
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1060: Console/WiiWare/VC
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1070: Console/XBOX 360 DLC
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1040: Console/Xbox
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1050: Console/Xbox 360
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1140: Console/Xbox One
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 6060: XXX/ImgSet
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 7020: Books/Ebook
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1035: Console/Switch
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1060: Console/WiiWare/VC
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1070: Console/XBOX 360 DLC
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1040: Console/Xbox
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1050: Console/Xbox 360
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 1140: Console/Xbox One
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 6060: XXX/ImgSet
[Error] Cardigann: CardigannIndexer (miatrix-yml): invalid Torznab category for id 7020: Books/Ebook
```

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (Qstick) [inline .vscode/launch.json:1]: Git ignore this folder

File: NewznabToCardigannYML.ps1 [hunk 3]
```diff
@@ -0,0 +1,68 @@
+<#
+    .SYNOPSIS
+        Name: Invoke-NewznabToCardigannYML.ps1
```
Comment (Qstick) [inline NewznabToCardigannYML.ps1:3]: `scripts` folder

File: newznabcats.txt [hunk 4]
```diff
@@ -0,0 +1,75 @@
+---- --------------------
+id   title
+---- --------------------
+1000 Console
+1010 Console/NDS
+1020 Console/PSP
```
Comment (Qstick) [inline newznabcats.txt:6]: This for the script? If so stick in the same folder

**Review by Qstick — CHANGES_REQUESTED** [review]

File: NewznabToCardigannYML.ps1 (hunk 3 above)
Comment (bakerboy448) [reply NewznabToCardigannYML.ps1]: updated x3

File: definitions/v4/miatrix.yml [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    - {id: 7040, cat: Books/Technical, descr: Books/Technical}
+    - {id: 8000, cat: Other, descr: Other}
+    - {id: 8020, cat: Other/Hashed, descr: Other/Hashed}
+    - {id: 8010, cat: Other/Misc, descr: Other/Misc}
+
+  modes:
+    # Be sure to update the search inputs
+    search: [q] # group is supported by the indexer but not Cardigann/Prowlarr
+    tv-search: [q, rid, tvdbid, tvmazeid, season, ep]
+    movie-search: [q, imdbid] # genre is supported by the indexer but not Cardigann/Prowlarr
```
Comment (Qstick) [inline definitions/v4/miatrix.yml:90]: We can likely add these extra features pretty easily on our side as part of the PR on the other repo.

File: definitions/v4/drunkenslug.yml [hunk 6]
```diff
@@ -0,0 +1,164 @@
+---
+id: drunkenslug-yml
+name: DrunkenSlug (YML)
```
Comment (Qstick) [inline definitions/v4/drunkenslug.yml:3]: Would prefer to keep the names clean without `(YML)` if we can. It means nothing to the average user

File: definitions/v4/drunkenslug.yml [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+description: "DrunkenSlug is a Private Usenet Indexer"
+language: en-US
+type: private
+redirect: false
```
Comment (Qstick) [inline definitions/v4/drunkenslug.yml:7]: All of these should support redirecting, basically anything that can provide a download link that doesn't require auth. In this case all you need is an APIKEY in the query, thus any NEWZNAB indexer is eligible to allow redirect if user wants it.

**Review by Qstick — CHANGES_REQUESTED** [review]

File: definitions/v4/drunkenslug.yml [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+    - {id: 7020, cat: Books/EBook, desc: Books/Ebook}
+    - {id: 7060, cat: Books/Foreign, desc: Books/Foreign}
+    - {id: 7010, cat: Books/Mags, desc: Books/Magazines}
+    - {id: 7999, cat: Books/Other, desc: Books/Other}
+    - {id: 7040, cat: Books/Technical, desc: Books/Technical}
+
+  modes:
+    # Be sure to update the search inputs
+    search: [q]
+    tv-search: [q,tvdbid,rid,tvmazeid,imdbid,season,ep] # vid, traktid, tmdbid are supported by the indexer but not Cardigann/Prowlarr
```
Comment (Qstick) [inline definitions/v4/drunkenslug.yml:84]: `traktid` is already supported for tv

**Comment by bakerboy448** [comment]
- [ ] - Note for me: cardigann now supports genre (movie and music) and TMDbID (TV)

**Comment by JigSawFr** [comment]
> * [ ]  - Note for me: cardigann now supports genre (movie and music) and TMDbID (TV)

Adding TMDBID for tv-series searches will be great !
Available for v4 version ?

**Comment by bakerboy448** [comment]
The additional ID based support already added to Dev

**Comment by JigSawFr** [comment]
> The additional ID based support already added to Dev

Yes, but it will be only supported in v4 of indexers ?
Or we can already update v3 definitions?

**Comment by bakerboy448** [comment]
Theoretically can update the v3 definitions, but @Qstick how do we want to handle those? Just v3 and hold back until we cut a dev?

**Comment by Qstick** [comment]
I'd have to see how an old build handles it. If it just ignore the extra params then it's likely fine to push it as v3. If it blows up, then v4 it is

**Review by Qstick — APPROVED** [review]

**Pull request merged**